status/controller.go

```diff
@@ -275,21 +275,23 @@ func (c *Controller[T]) reconcile(ctx context.Context, req reconcile.Request, o
 	}
 
 	for _, observedCondition := range observedConditions.List() {
-		if currentCondition := currentConditions.Get(observedCondition.Type); currentCondition == nil || currentCondition.Status != observedCondition.Status || currentCondition.Reason != observedCondition.Reason || !maps.Equal(c.toAdditionalGaugeMetricLabels(o), observedGaugeLabels) {
-			c.deletePartialMatchGaugeMetric(c.ConditionCount, ConditionCount, lo.Assign(map[string]string{
+		currentCondition := currentConditions.Get(observedCondition.Type)
+		conditionChanged := currentCondition == nil || currentCondition.Status != observedCondition.Status || currentCondition.Reason != observedCondition.Reason
+		if conditionChanged || !maps.Equal(c.toAdditionalGaugeMetricLabels(o), observedGaugeLabels) {
```

Comment:
nit: So technically what you can do here is you can only apply the additionalGaugeMetricLabels to the delete when you actually need to include them in the delete -- basically, add them when they change, don't add them when they don't and I think everything should work as it should